tests/test.rs

```diff
@@ -329,10 +353,24 @@ fn set_admin_bad_signature() {
     let admin1_id = Identifier::Ed25519(admin1.public.to_bytes());
     let admin2_id = Identifier::Ed25519(admin2.public.to_bytes());
 
-    token.initialize(&admin1_id);
+    token.initialize(&admin1_id, 10, "name", "symbol");
 
     let mut signature: [u8; 64] = vec![0; 64].as_slice().try_into().unwrap();
     thread_rng().fill_bytes(&mut signature);
     let auth = external::Authorization::Ed25519(signature);
     external::set_admin(&mut token.0, &token.1, &auth, &admin2_id);
 }
+
+#[test]
+#[should_panic(expected = "Decimal must fit in a u8")]
+fn decimal_is_over_max() {
+    let e = Env::with_empty_recording_storage();
+    let contract_id = generate_contract_id();
+    external::register_test_contract(&e, &contract_id);
+    let mut token = Token(e, contract_id.clone());
+
+    let admin1 = generate_keypair();
+    let admin1_id = Identifier::Ed25519(admin1.public.to_bytes());
+
+    token.initialize(&admin1_id, u32::from(u8::MAX) + 1, "name", "symbol");
```

Comment:
I'm guessing this is why you made the `initialize` external function take `u32`. Better to just directly call the contract with bad data, rather than make the external interface easy to misuse.

Reply:
Done

Reply:
@sisuresh I see you already did this, but in retrospect I think my judgment was wrong for a couple of reasons:

1. This won't really be possible with https://github.com/stellar/rs-stellar-contract-sdk/pull/316
2. It should be easy to test any possible input using these external functions without calling `invoke_function` directly. Normally you wouldn't want to call a function with a value you know is bad, but since this isn't the real interface to the function it actually makes sense to allow the bad value.

How to proceed is up to you. If you don't want to revert my bad idea, I'll gladly do it for you when I do the follow-on work to https://github.com/stellar/rs-stellar-contract-sdk/pull/316. If you do want to revert it, feel free. Just let me know so I know whether I should merge this PR or not.

Sorry for the bad recommendation.

Reply:
I reverted this change.